stdlib/LinearAlgebra/src/givens.jl

```diff
@@ -250,20 +250,23 @@ function givensAlgorithm(f::Complex{T}, g::Complex{T}) where T<:AbstractFloat
     return cs, sn, r
 end
 
-# enable for unitful quantities
-function givensAlgorithm(f::T, g::T) where T
-    fs = f / oneunit(T)
-    gs = g / oneunit(T)
-    typeof(fs) === T && typeof(gs) === T &&
-    !isa(fs, Union{AbstractFloat,Complex{<:AbstractFloat}}) &&
-    throw(MethodError(givensAlgorithm, (fs, gs)))
-
-    c, s, r = givensAlgorithm(fs, gs)
-    return c, s, r * oneunit(T)
+# From Janovská, D., & Opfer, G. (2003). Givens’ Transformation Applied to Quaternion
+# Valued Vectors. BIT Numerical Mathematics, 43(5), 991–1002.
+# doi:10.1023/b:bitn.0000014561.58141.2c
+function givensAlgorithm(f::Number, g::Number)
+    nrm = hypot(f, g)
+    c = abs(f) / nrm
+    s, u = if iszero(f)
+        -one(first(promote(f, g))), -g
```

Comment:
Is there a reason against using `one(promote_type(typeof(f), typeof(g)))` here? Quick testing indicates a performance difference:
```julia

julia> f = big(2)
2

julia> g = big(3.0)
3.0

julia> @btime one(promote_type(typeof($f), typeof($g)))
  32.062 ns (2 allocations: 104 bytes)
1.0

julia> @btime one(first(promote($f, $g)))
  68.061 ns (4 allocations: 208 bytes)
1.0
```

Reply:
promote always returns a concrete type, while promote_type might return a type join

Reply:
I see. Thank you